src/main/java/io/quarkus/registry/app/model/PlatformRelease.java

```diff
@@ -137,17 +147,42 @@ public static Optional<PlatformRelease> findByNaturalKey(PlatformStream platform
     }
 
     public static List<PlatformRelease> findLatest(String quarkusCore) {
-        EntityManager entityManager = getEntityManager();
+        EntityManager em = getEntityManager();
         TypedQuery<PlatformRelease> query;
-        if (quarkusCore != null && !quarkusCore.isBlank()) {
-            query = entityManager.createNamedQuery("PlatformRelease.findLatestByQuarkusCoreVersion", PlatformRelease.class)
+        boolean withQuarkusCore = quarkusCore != null && !quarkusCore.isBlank();
+        if (withQuarkusCore) {
+            query = em.createNamedQuery("PlatformRelease.findLatestByQuarkusCoreVersion", PlatformRelease.class)
                     .setParameter(1, quarkusCore);
         } else {
-            query = entityManager.createNamedQuery("PlatformRelease.findLatest", PlatformRelease.class);
+            query = em.createNamedQuery("PlatformRelease.findLatest", PlatformRelease.class);
+        }
+        // Check if we can perform this directly in the SQL
+        // We need the top 3 releases for each platform
```

Comment:
Looks like pinned releases are not included in these 3 but added to those?

Reply:
No, that's the purpose of pinning - to always include them. Ok, thanks for clarifying.